pika/adapters/select_connection.py

```diff
@@ -276,7 +260,9 @@ def process_timeouts(self):
         # were only run periodically.
         for t in sorted(to_run, key=itemgetter('deadline')):
             t['callback']()
-            del self._timeouts[hash(frozenset(t.items()))]
+            k = hash(frozenset(t.items()))
+            if k in self._timeouts:
```

Comment:
Another surprising aspect of the legacy implementation is the unnecessary re-computation of the key. The hash key is available in self._timeouts, so the `to_run = ...` statement could also save the key along with `timer` as a tuple in order to avoid the re-computation here.